Update README.md (#1376)
SoftwareAG/c8y-docs: README.md

# Cumulocity IoT documentation website

<https://cumulocity.com/guides> - releases 10.18 and older
<https://cumulocity.com/docs> - CD and yearly releases

**Built with [Hugo](https://gohugo.io/)** (version 0.92.1)

## Development environment

- Clone the repository.
- Install Hugo by using snap `snap install hugo --channel=extended`.

> - Open a terminal window in the project directory and type `hugo server`. Specify the subdirectory that contains your content, for example http://localhost:1313/guides/users-guide/. Specifying only http://localhost:1313/guides results in a 404 error.

Reviewer: /users-guide/ no longer exists in CD/yearly.
we should probably use an example of the new structure here.
same for /guides, which is /docs in the future.

Author: But we still have the release branches in the repo which is why I left the /guide notation for now.

Reviewer: alright. we should change it once the new structure becomes the default then.

## Structure for /guides (for releases 10.18 and older)

The Cumulocity IoT documentation website architecture has the following structure:

- 1st level  - for example, *<http://cumulocity.com/guides/users-guide/>*
  - 2nd level - renders as a single page, for example *<http://cumulocity.com/guides/users-guide/administration/>*
    - Anchor section - anchor tag in the subsection page, for example *<http://cumulocity.com/guides//users-guide/administration/#managing-permissions>*

The architecture is built with a mix of front matter and directory structure.

## Structure - /docs (for CD and yearly releases)

The Cumulocity IoT documentation website architecture has the following structure:

- 1st level  - for example, *<http://cumulocity.com/docs/platform_administration>*
  - 2nd level - for example *<http://cumulocity.com/docs/platform_administration/standard-tenant/>*
    - 3rd level - renders as a single page, for example *<http://cumulocity.com/docs/standard-tenant/managing-permissions>*
      - Anchor section - anchor tag in the subsection page, for example *<http://cumulocity.com/docs/standard-tenant/managing-permissions/#global-roles>*

> The architecture is built with a mix of front matter and directory structure.

Reviewer: nitpicky, but an architecture is the building blueprint, not that which is actually being built.
The website is built :)

(also above)

Author: This text originally has been created by Carlos :-). But you are right, will change it.

## Adding content

### 1. Add a new section

> In the new documentation structure (/docs) adding level 1 or level 2 sections is restricted to the admin users of the c8y-docs repo.

Reviewer: is it technically, or is it something we filter by PR, if it ever occurs?
and is it relevant in terms of how we communicate it?

Author: It is not technically, but nobody needs to know :-). We should not have anyone adding tiles without consulting us.

Reviewer: alright

### 3. Add a subsection

Inside the newly created directory create a Markdown file with the name you wish to use as a URL — for example `introduction.md` with the following front matter:

```yaml
---
title: Introduction to Cumulocity # the page title
layout: bundle # don't change
weight: 10 # set the position of the page within the section in ascending order
aliases: # if needed, add the redirects here, otherwise remove this
  - /concepts/introduction-to-cumulocity/
  - /concepts/introduction-to-cumulocity.html
---
# add optional content as Markdown
Cumulocity gives you very fast visibility and control over your remote assets, be these houses, cars, machines or any other assets that you need to manage.
```

If you're looking to have a short page without anchors, you're good to go, but if you want to add multiple subsections with anchors then proceed to the next step.

When adding multiple subsections, the content provided in this file will be rendered as a lead in the top of the page.

### 4. Add blocks of content with anchors to a page

To display multiple blocks of content and provide anchor links to display in the navigator, you'll have to follow these steps:

> #### 1. Add a directory with the exact same name as the Markdown file adding the suffix `-bundle`, e.g. `introduction-bundle`.

Reviewer: ```suggestion
#### 1. Add a directory with the exact same name as the Markdown file adding the suffix `-bundle`, for example, `introduction-bundle`
```

also very long as a heading, and ideally should then not contain numbering or more markup
why can it not be an enumeration?

Author: Again, this has been written by Carlos. Will change that.

Media should be added to `/static/images/`. Add a new directory if none of the available suits your needs.

Keep all file names URL-friendly (lowercase, no special characters, and no empty spaces).

> To use the images in your pages, just add the relative path, for example `![image title](/images/<directory name>/<file name>)`.

Reviewer: ```suggestion
To use the images in your pages, just add the relative path, for example: `![image title](/images/<directory name>/<file name>)`.
```

## Redirects

Redirects must be processed through aliases. Add aliases as an array, and make sure to remove `/docs` or `/guides` out of the URL. Check the following example:

```yaml
---
title: Introduction to Cumulocity
layout: bundle
weight: 10
aliases:
  - /concepts/introduction-to-cumulocity/
  - /concepts/introduction-to-cumulocity.html
# this will redirect cumulocity.com/docs/concepts/introduction-to-cumulocity/
# to cumulocity.com/docs/concepts/introduction
---
```

## Deploying to cumulocity.com/guides

> Cumulocity provides documentation for multiple releases. You must create a release branch for every public release, for example `release/r10.16.0`. When creating these branches follow the next steps:

Reviewer 2: ```suggestion
Cumulocity provides documentation for multiple releases. You must create a release branch for every public release, for example: `release/r10.16.0`. When creating these branches follow the next steps:
```

> Cumulocity provides documentation for multiple releases, for that you'll have to create a release branch for every public release, for example `release/r10.16.0`. When creating these branches follow the next steps:

Reviewer: ```suggestion
Cumulocity provides documentation for multiple releases. You must create a release branch for every public release, for example `release/r10.16.0`. When creating these branches follow the next steps:
```
